### PR TITLE
More flexiable nexe footer search range to fix digital sign binary fail to launch issue

### PR DESCRIPTION
**Pull request by foxever**
Fix Windows binary break after digital sign.
Prebuilt binary still works with this approach.
Credit to  @tamasszarka https://github.com/nexe/nexe/issues/372#issuecomment-438209571

### CODE REVIEW

File: src/patches/boot-nexe.ts [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
   fd = fs.openSync(process.execPath, 'r'),
-  stat = fs.statSync(process.execPath),
-  footer = Buffer.alloc(32, 0)
+  stat = fs.statSync(process.execPath)
 
-fs.readSync(fd, footer, 0, 32, stat.size - 32)
-
-if (!footer.slice(0, 16).equals(Buffer.from('<nexe~~sentinel>'))) {
+// need to find the real nexe footer for validation, but the
+// signing process adds extra bytes to the end of the file
```
Comment (calebboyd) [inline src/patches/boot-nexe.ts]: You can toss these comments, the code intent is pretty clear

File: src/patches/boot-nexe.ts [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+const tailSize = Math.min(stat.size, 16000)
+const tailBuf = Buffer.from(Array(tailSize))
```
Comment (calebboyd) [inline src/patches/boot-nexe.ts]: combine const statements (should be linted, sorry), naming nit
```
const tailSize = Math.min(stat.size, 16000),
  tailWindow = Buffer.alloc(tailSize)
```

File: src/patches/boot-nexe.ts [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
+fs.readSync(fd, tailBuf, 0, tailSize, stat.size - tailSize)
+const footerPos = tailBuf.indexOf('<nexe~~sentinel>')
```
Comment (calebboyd) [inline src/patches/boot-nexe.ts]: `footerPosition`

File: src/patches/boot-nexe.ts [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
+if (footerPos == -1) {
   throw 'Invalid Nexe binary'
 }
 
+const footer = tailBuf.slice(footerPos, footerPos + 32)
+
 const contentSize = footer.readDoubleLE(16),
```
Comment (calebboyd) [inline src/patches/boot-nexe.ts]: combine with previous const (`,`)

**Comment by Lange** [comment]
I am still experiencing https://github.com/nexe/nexe/issues/372 after signing my Windows binary, even with nexe v3.1.0. Is there anything special I need to do to get this to work? Any particular build flags I need to use?

**Comment by calebboyd** [comment]
@Lange That is correct. The prebuilds haven't been updated yet. I'm hoping to get them running soon but don't have an eta at the moment.

**Comment by Lange** [comment]
@calebboyd Got it, makes total sense, thanks for clarifying! I shall wait patiently and use `--build` in the meantime.

**Comment by ricardopolo** [comment]
Please confirm here when they are updated!! Thank you!!

**Comment by tamasszarka** [comment]
Are you still planning to update the prebuilds? Our CI system spends an extra 25minutes multiple times a day due to the --build flag :) Do you think we can create local prebuilds for ourself somehow? Thanks a lot!

**Comment by tamasszarka** [comment]
@calebboyd sorry that I keep pinging here. I'm really desperate about getting prebuilds for v3.0.3+.
I'd be more than happy to help generating the official ones too, if you provide some guidance. Can you help me?

**Comment by calebboyd** [comment]
What version are you needing?

**Comment by ricardopolo** [comment]
@calebboyd I need 10.15.3 😄

**Comment by tamasszarka** [comment]
@calebboyd I'd need 10.12.0. Thanks a lot!

**Comment by calebboyd** [comment]
Builds incoming. They'll populate over the next few days

**Comment by tamasszarka** [comment]
@calebboyd Will the new builds appear here? https://github.com/nexe/nexe/releases

**Pull request closed without merging**